controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1755,18 +1795,28 @@ func (r *BareMetalHostReconciler) createHostFirmwareComponents(info *reconcileIn
 
 			// Set bmh as owner, this makes sure the resource is deleted when bmh is deleted
 			if err = controllerutil.SetControllerReference(info.host, hfc, r.Scheme()); err != nil {
-				return errors.Wrap(err, "could not set bmh as controller")
+				return errors.Wrap(err, "could not set bmh as controller for hostFirmwareComponents")
 			}
 			if err = r.Create(info.ctx, hfc); err != nil {
 				return errors.Wrap(err, "failure creating hostFirmwareComponents resource")
 			}
 
 			info.log.Info("created new hostFirmwareComponents resource")
-		} else {
-			// Error reading the object
-			return errors.Wrap(err, "could not load hostFirmwareComponents resource")
+			return nil
 		}
+		// Error reading the object
+		return errors.Wrap(err, "could not load hostFirmwareComponents resource")
 	}
+	// Necessary in case the CRD is created manually.
+	err := controllerutil.SetControllerReference(info.host, hfc, r.Scheme())
+	if err == nil {
+		if err := r.Update(info.ctx, hfc); err != nil {
```

Comment:
I still feel somewhat uneasy calling Update every time. I see other places use `if !ownerReferenceExists(...)`, maybe use that as well?

Reply:
oh like we have for DataImage, I will give a try and test in my setup